src/background/background.ts

```diff
@@ -1,4 +1,151 @@
+EDUCATION
+BSc. in Computer Science, Sorbonne Université - Campus Pierre et Marie Curie— Paris
+2012 - 2015
+Diploma in Audio Engineering, SAE Institute — Paris
+2015 - 2017
+Online courses
+Ongoing
+Continuously enhancing skills through Udemy courses, including:
+Beginning C++ Programming - From Beginner to Beyond
+The Complete JavaScript Course
+React - The Complete Guide
+Node.js, Express, MongoDB & More
+The Ultimate MySQL Bootcamp: Go from SQL Beginner to Expert
+React & TypeScript Chrome Extension Development
+JavaScript Algorithms and Data Structures Masterclass
+Software Architecture & Design of Modern Large Scale Systems
+PREVIOUS EXPERIENCE
+
+Self Employed, Tel Aviv — Sound Designer
+March 2017 - January 2023
+As the ex-owner of a top-notch sound studio in Israel, based in Tel Aviv, I led the way on audio design projects for big video production companies. Working with huge names like McDonald's, AMPM, Samsung, Strauss, Tnuva, and providing sound design solutions for big tech companies such as Wix, SAP, Fiverr, Tabula, Simply, Viber, and others. I consistently delivered top-quality work.
+Successfully sold the business for a profit.
+
+
+SKILLS
+Deep understanding of Vanilla Javascript
+fluency with relational and NoSQL databases(MySQL, MongoDB)
+Proficient in implementing best practices for designing and maintaining APIs.
+Experienced in applying responsive design principles for optimal user experience.
+Experience with Git and GitHub
+Experienced in Chrome extension development
+Experienced in Vs Code extension development
+PROGRAMING LANGUAGES
+Javascript, Typescript, C++
+FRAMEWORKS
+Express, ReactJS
+`;
+// Global states - Chrome Storage ---------------------
+
+type GlobalStateKey = 'query'; // we might need:  'jobDescription' | 'currentUrl'  | 'jobPostId';
+
+function setState(key: GlobalStateKey, value: any) {
+  chrome.storage.sync.set({ [key]: value }, () => {
+    console.log(`Data saved - key: ${key} and value: ${value}`);
+  });
+}
+
+function getState(key: GlobalStateKey) {
+  return new Promise((resolve, reject) => {
+    chrome.storage.sync.get(key, (result) => {
+      if (chrome.runtime.lastError) {
+        reject(chrome.runtime.lastError);
+      } else {
+        resolve(result[key]);
+      }
+    });
+  });
+}
+
+function sendToContentScript(message: Message) {
+  chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+    // Send a message to the content script of the active tab
+    chrome.tabs.sendMessage(
+      tabs[0].id,
+      { message },
+      async (response: Message) => {
+        console.log('Response from content script:', response);
+        if (response.type === 'catch-job-description') {
+          const query = getQuery(response.message);
+          setState('query', query);
+        }
+      }
+    );
+  });
+}
+
+function getQuery(jobDescription: string) {
+  return `You are a smart assistant to career advisors at the Harvard Extension School. Your take is to rewrite
```

Comment:
save constants seperate from the logic

Reply:
Done, i left some of the text in the function because I need the jobDescription argument to return the query